states/huntsman/ready.py

```diff
@@ -1,3 +1,27 @@
+import time
+from pocs.utils import error
+
+
+def prepare_cameras(pocs, sleep=60, max_attempts=5):
+    """
+    Make sure cameras are ready before starting states.
+    """
+    pocs.logger.debug('Preparing cameras before leaving ready state.')
+    cameras = pocs.observatory.cameras.values()
+
+    # Make sure camera cooling is enabled
+    for cam in cameras:
+        if cam.is_cooled_camera:
+            cam.cooling_enabled = True
+
+    # Wait for cameras to be ready
+    for i in range(max_attempts):
+        if all([cam.is_ready() for cam in cameras]):
+            return
+        time.sleep(sleep)
+    raise error.PanError('Timeout while waiting for cameras to become ready from ready state.')
```

Comment:
Not sure I agree with that one. Until we have some way to notify us of any errors (e.g. cameras not working) I would rather an error be raised immediately so we realise what's happened.

Reply:
You could argue that one the other way around. If we don't have some way to notify us of errors then it's even more important for Huntsman to just deactivate a problematic camera and carries on observing without it, because the alternative is Huntsman not doing any more observing until someone checks on it.

For errors that don't compromise safety (e.g. camera or focuser errors) I do think that Huntsman should deactivate the effected system and carry on observing if possible.

Reply:
I agree that might be a reasonable approach during normal observing. But we are still in the early phases of commissioning and I would like keep hardware errors as self-evident as possible over this time. Maybe it could be implemented at a later phase in commissioning?